compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -768,7 +767,7 @@ impl<'tcx> Inliner<'tcx> {
         //
         // and the vector is `[closure_ref, tmp0, tmp1, tmp2]`.
         if callsite.fn_sig.abi() == Abi::RustCall && callee_body.spread_arg.is_none() {
-            let mut args = args.into_iter();
+            let mut args = Vec::from(args).into_iter();
```

Comment:
`Box<[T]>` implements `IntoIterator` since https://github.com/rust-lang/rust/pull/124097 -- why can't you just use that?

Reply:
Oh, might not be on beta yet. Pls leave a fixme on this callsite and others.

Reply:
Oh wait, it's because we literally just don't select the `Box<[]>::into_iter` method when doing `.` dispatch in edition 2021. Yeah, could be `// FIXME(edition_2024):` or something.